contracts/Liquidator/Liquidator.sol

```diff
@@ -260,23 +348,80 @@ contract Liquidator is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable {
         vai.safeApprove(address(vaiController), 0);
         vai.safeApprove(address(vaiController), repayAmount);
 
-        (uint err, ) = vaiController.liquidateVAI(borrower, repayAmount, vTokenCollateral);
+        (uint256 err, ) = vaiController.liquidateVAI(borrower, repayAmount, vTokenCollateral);
         requireNoError(err);
     }
 
-    /// @dev Splits the received vTokens between the liquidator and treasury.
+    /// @dev Distribute seized collateral between liquidator and protocol share reserve
     function _distributeLiquidationIncentive(
         IVToken vTokenCollateral,
-        uint256 siezedAmount
+        uint256 seizedAmount
     ) internal returns (uint256 ours, uint256 theirs) {
-        (ours, theirs) = _splitLiquidationIncentive(siezedAmount);
+        (ours, theirs) = _splitLiquidationIncentive(seizedAmount);
         if (!vTokenCollateral.transfer(msg.sender, theirs)) {
             revert VTokenTransferFailed(address(this), msg.sender, theirs);
         }
-        if (!vTokenCollateral.transfer(treasury, ours)) {
-            revert VTokenTransferFailed(address(this), treasury, ours);
+
+        if (ours > 0 && !_redeemUnderlying(address(vTokenCollateral), ours)) {
+            // Check if asset is already present in pendingRedeem array
+            uint256 index;
+            for (index; index < pendingRedeem.length; ) {
```

Comment:
This loop has a maximum of `comptroller.maxAssets()` iterations, right?

Reply:
This loop prevents redundant addition of asset to array for future redeem if it fails in current transaction, in worst case if each and every market redeem fails (may be action paused, may be unavailibility of liquidity) it can go up to `maxSupportedAssets -1`